lib/puppet/pops/loaders.rb

```diff
@@ -306,9 +306,18 @@ def create_loader_with_all_modules_visible(from_module_data)
 
     def create_loader_with_only_dependencies_visible(from_module_data)
       if from_module_data.unmet_dependencies?
-        Puppet.warning("ModuleLoader: module '#{from_module_data.name}' has unresolved dependencies"+
-          " - it will only see those that are resolved."+
-          " Use 'puppet module list --tree' to see information about modules")
+        if Puppet[:strict] != :off
+          msg = "ModuleLoader: module '#{from_module_data.name}' has unresolved dependencies"+
+              " - it will only see those that are resolved."+
+              " Use 'puppet module list --tree' to see information about modules"
```

Comment:
You don't need the '+' operator in Ruby, it will concatenate strings for you in the parsing phase. With + it will instead copy/concatenate at runtime.